templater/kustomize/kustomize.go

```diff
@@ -8,79 +8,95 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
-type KustomizeProviderOptions struct {
-	Ref               string
+type KustomizeUpdate struct {
 	KustomizationFile string
 	Image             string
 	NewImage          string
 	NewTag            string
 }
 
-func UpdateKustomization(repository targets.Repository, options KustomizeProviderOptions) (targets.FileList, error) {
-	file, err := repository.Get(options.KustomizationFile, options.Ref)
-	if err != nil {
-		return nil, fmt.Errorf("could not retrieve values.yaml from repository: %w", err)
-	}
+type KustomizeProviderOptions struct {
+	Ref     string
+	Updates []KustomizeUpdate
+}
 
-	values := make(map[string]interface{})
-	if err := yaml.Unmarshal(file, values); err != nil {
-		return nil, fmt.Errorf("could not parse values.yaml: %w", err)
-	}
+func UpdateKustomization(repository targets.Repository, options KustomizeProviderOptions) (targets.FileList, error) {
+	original := make(map[string][]byte)
+	files := make(map[string]map[string]interface{})
+	for _, update := range options.Updates {
+		if _, ok := files[update.KustomizationFile]; !ok {
+			file, err := repository.Get(update.KustomizationFile, options.Ref)
+			if err != nil {
+				return nil, fmt.Errorf("could not retrieve %s from repository: %w", update.KustomizationFile, err)
+			}
 
-	images, ok := values["images"]
-	if !ok {
-		images = []interface{}{}
-	}
+			original[update.KustomizationFile] = file
+			files[update.KustomizationFile] = make(map[string]interface{})
+			if err := yaml.Unmarshal(file, files[update.KustomizationFile]); err != nil {
+				return nil, fmt.Errorf("could not parse YAML file %s: %w", update.KustomizationFile, err)
+			}
+		}
 
-	// Make sure existing value is an array
-	imageList, ok := images.([]interface{})
-	if !ok {
-		return nil, fmt.Errorf("kustomization file .images field is not an array")
-	}
+		values := files[update.KustomizationFile]
+		images, ok := values["images"]
+		if !ok {
+			images = []interface{}{}
+		}
 
-	// Check for existing entries
-	found := false
-	for index := range imageList {
-		current, ok := imageList[index].(map[string]interface{})
+		// Make sure existing value is an array
+		imageList, ok := images.([]interface{})
```

Comment:
nitpick: could we type it better? I'm always scared by raw `interface{}` types as they can hide bugs

Reply:
We can't, it will only match `[]interface{}` (or `[]any`, should probably change to that) if it's an array, this is how YAML/Reflection work, not my choice :C

Reply:
Argh, fine then